cogeo_mosaic/backends/gs.py

```diff
@@ -0,0 +1,102 @@
+"""cogeo-mosaic Google Cloud Storage backend."""
+
+import json
+from typing import Any
+from urllib.parse import urlparse
+
+import attr
+from cachetools import TTLCache, cached
+from cachetools.keys import hashkey
+
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import _compress_gz_json, _decompress_gz
+from cogeo_mosaic.cache import cache_config
+from cogeo_mosaic.errors import _HTTP_EXCEPTIONS, MosaicError, MosaicExistsError
+from cogeo_mosaic.mosaic import MosaicJSON
+
+try:
+    from google.cloud.storage import Client as gcp_session
+    from google.auth.exceptions import GoogleAuthError
+except ImportError:  # pragma: nocover
+    gcp_session = None  # type: ignore
```

Comment:
Should you just raise the `ImportError` here?

Reply:
no because it's an optional dependency.

we do the same in https://github.com/developmentseed/cogeo-mosaic/blob/master/cogeo_mosaic/backends/dynamodb.py#L31-L38

and then raise an error later if you try to use it https://github.com/developmentseed/cogeo-mosaic/blob/master/cogeo_mosaic/backends/dynamodb.py#L54-L63

Reply:
oh duh 🤦‍♂️  because this file is imported from `__init__.py` so it's run even if not using this backend

Reply:
yep